### PR TITLE
✅ Improve test coverage

### PR DESCRIPTION
**Pull request by lig**
## Change Summary

Various test coverage improvements

## Related issue number

None

## Checklist

* [x] Unit tests for the changes exist
* [x] Tests pass on CI and coverage remains at 100%
* [ ] Documentation reflects the changes where applicable
* [ ] `changes/<pull request or issue id>-<github username>.md` file added describing change
  (see [changes/README.md](https://github.com/samuelcolvin/pydantic/blob/master/changes/README.md) for details)
* [x] My PR is ready to review, **please add a comment including the phrase "please review" to assign reviewers**

Selected Reviewer: @adriangb

### CODE REVIEW

**Comment by lig** [comment]
please, review

File: pydantic/_internal/_generate_schema.py [hunk 1]
```diff
@@ -991,7 +991,7 @@ def _pattern_schema(self, pattern_type: Any) -> core_schema.CoreSchema:
                 _validators.pattern_bytes_validator, serialization=ser, metadata=metadata
             )
         else:
-            raise PydanticSchemaGenerationError(f'Unable to generate pydantic-core schema for {pattern_type!r}.')
+            raise PydanticSchemaGenerationError(f"Unable to generate pydantic-core schema for '{pattern_type!r}'.")
```
Comment (samuelcolvin) [inline pydantic/_internal/_generate_schema.py:994]: you don't need the quotes here, you already have the `!r`.

**Review by samuelcolvin — COMMENTED** [review]
otherwise LGTM.

Please update.

File: pydantic/_internal/_generate_schema.py (hunk 1 above)
Comment (lig) [reply pydantic/_internal/_generate_schema.py:994]: Thank you! That was before you've pointed me to this the previous time.

**Comment by lig** [comment]
please review

File: tests/test_fields.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    """This tests that `FieldInfo.from_field` raises an error if passed the `annotation` kwarg.
+
+    At the time of writing this test there is no way `FieldInfo.from_field` could receive the `annotation` kwarg from
+    anywhere inside Pydantic code. However, it is possible that this API is still being in use by applications and
+    third-party tools.
+    """
+    with pytest.raises(TypeError) as e:
+        fields.FieldInfo.from_field(annotation=())
+
+    assert e.value.args == ('"annotation" is not permitted as a Field keyword argument',)
```
Comment (adriangb) [inline tests/test_fields.py:20]: If we're going to just raise an error, I'm not sure there's much value in providing a custom error if we can't offer a solution / link to workaround.

**Review by adriangb — APPROVED** [review]

File: tests/test_internal.py [hunk 3]
```diff
@@ -117,3 +120,22 @@
...
+
+    obj = Obj()
+
+    assert str(devtools.debug.format(obj)).split('\n')[1:] == [
+        '    Obj(',
+        '        int_attr=42,',
+        "        str_attr='Marvin',",
+        '    ) (Obj)',
+    ]
+    assert list(obj.__rich_repr__()) == [('int_attr', 42), ('str_attr', 'Marvin')]
```
[the remark this answers is not in the document]
Comment (lig) [reply tests/test_internal.py]: I'm testing here through `devtools` and `rich` facing APIs. The function I'm actually covering lays deeper. I've renamed the test to address that this tests integrations.

**Comment by lig** [comment]
added comments.
please review.

**Review by adriangb — APPROVED** [review]

**Pull request merged**